includes/abstracts/abstract-batch.php

```diff
@@ -136,8 +143,47 @@
 	 */
 	public function get_results() {
 		$this->args = wp_parse_args( $this->args, $this->default_args );
+		$results = $this->batch_get_results();
 		$this->calculate_offset();
-		return $this->batch_get_results();
+		return $results;
+	}
+
+	/**
+	 * Set the total number of results
+	 *
+	 * Uses a number passed from the client to the server and compares it to the total objects
+	 * pulled by the latest query. If the dataset is larger, we increase the total_num_results number.
+	 * Otherwise, keep it at the original (to acount for deletion / changes).
+	 *
+	 * @param int $total_from_query Total number of results from latest query.
+	 */
+	public function set_total_num_results( $total_from_query ) {
+		// If this is past step 1, the client is passing back the total number of results.
+		// This accounts for deletion / descructive actions to the data.
+		$total_from_request = isset( $_POST['total_num_results'] ) ? absint( $_POST['total_num_results'] ) : 0; // Input var okay.
```

Comment:
Yea, that makes sense. I just don't know where to move it. We could put the global in it's own function, but that will need to be called somewhere either way, then passed down through to the function. Open to suggestions, I'm pretty deep in this code so an outside eye is always welcome 😃